### PR TITLE
add base class for seqlastin/max/average layer

### PR DESCRIPTION
**Pull request by luotao1**

### CODE REVIEW

File: paddle/gserver/layers/PoolLayer.cpp [hunk 1]
```diff
@@ -66,4 +65,70 @@ Layer* PoolLayer::create(const LayerConfig& config) {
   }
 }
 
+REGISTER_LAYER(seqpool, SequencePoolLayer);
+
+bool SequencePoolLayer::init(const LayerMap& layerMap,
```
Comment (emailweixu) [inline paddle/gserver/layers/PoolLayer.cpp]: Put this a different file: SequencePoolLayer(.h, .cpp)

File: paddle/gserver/layers/PoolLayer.cpp [hunk 2]
```diff
@@ -66,4 +65,70 @@ Layer* PoolLayer::create(const LayerConfig& config) {
   }
 }
 
+REGISTER_LAYER(seqpool, SequencePoolLayer);
```
Comment (emailweixu) [inline paddle/gserver/layers/PoolLayer.cpp]: Remove REGISTER_LAYER because this layer should not be used by user.

Comment (luotao1) [inline paddle/gserver/layers/PoolLayer.cpp]: Remove  REGISTER_LAYER in SequencePoolLayer.cpp

File: paddle/gserver/layers/PoolLayer.cpp (hunk 1 above)
Comment (luotao1) [inline paddle/gserver/layers/PoolLayer.cpp]: create a new SequencePoolLayer(.h, .cpp)

**Pull request merged**